Memorize/EmojiMemoryGameView.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  EmojiMemoryGameView.swift
+//  Memorize
+//
+//  Created by zhira on 7/8/24.
+//
+
+import SwiftUI
+
+struct EmojiMemoryGameView: View {
+    @ObservedObject var viewModel: EmojiMemoryGame
+
+    var body: some View {
+        VStack(spacing: 0) {
+            header
+            ScrollView {
+                cards
+                    .animation(.default, value: viewModel.cards)
+            }
+            Spacer()
+            Button(action: viewModel.startNewGame) {
+                Text("New Game")
+                    .fontWeight(.medium)
+            }
+            .buttonStyle(GrowingButton(color: viewModel.themeColor))
+        }
+        .padding()
+    }
+
+    private var header: some View {
+        HStack(alignment: .top) {
+            VStack {
+                Text("\(viewModel.themeName) \nMemorize!")
+                    .font(.largeTitle)
+                    .fontWeight(.semibold)
+                    .foregroundColor(viewModel.themeColor)
+            }
+            Spacer()
+            VStack {
+                Text("score")
+                    .font(.headline)
+                    .fontWeight(.bold)
+                    .foregroundColor(viewModel.themeColor)
+                Text("\(viewModel.score)")
+                    .font(.system(size: 64))
+                    .fontWeight(.semibold)
+                    .foregroundColor(viewModel.themeColor)
+            }
+        }
+    }
+
+    private var cards: some View {
+        LazyVGrid(
+            columns: [GridItem(.adaptive(minimum: 80), spacing: 0)],
+            spacing: 0
+        ) {
+            ForEach(viewModel.cards) { card in
+                CardView(card)
+                    .aspectRatio(2 / 3, contentMode: .fit)
+                    .padding(4)
+                    .onTapGesture {
+                        viewModel.choose(card)
+                    }
+            }
+        }
+        .foregroundColor(viewModel.themeColor)
+    }
+}
+
+#Preview {
+    EmojiMemoryGameView(viewModel: EmojiMemoryGame())
+}
+
+struct GrowingButton: ButtonStyle {
+    let color: Color
+
+    func makeBody(configuration: Configuration) -> some View {
+        configuration.label
+            .padding()
+            .background(color)
+            .foregroundStyle(.white)
+            .clipShape(Capsule())
+            .scaleEffect(configuration.isPressed ? 1.2 : 1)
+            .animation(.easeOut(duration: 0.5), value: configuration.isPressed)
+    }
+}
```

Comment:
It depends on context but, styles and thinks that could repeatable code, should in it's own file.